static/app/components/organizations/timeRangeSelector/dateRange.tsx

```diff
@@ -182,11 +188,15 @@ class BaseDateRange extends Component<Props, State> {
     // Subtract additional day  because we force the end date to be inclusive,
     // so when you pick Jan 1 the time becomes Jan 1 @ 23:59:59,
     // (or really, Jan 2 @ 00:00:00 - 1 second), while the start time is at 00:00
-    const minDate = getStartOfPeriodAgo(
-      'days',
-      (maxPickableDays ?? MAX_PICKABLE_DAYS) - 2
-    );
-    const maxDate = new Date();
+    let minDate = getStartOfPeriodAgo('days', (maxPickableDays ?? MAX_PICKABLE_DAYS) - 2);
+
+    let maxDate = new Date();
+
+    const isSameDay = moment(end).subtract(1, 'day').isSame(moment(start), 'day');
+    if (maxDateRange && isSameDay) {
```

Comment:
I'm confused, how does this work? If start and end are the same day, _then_ you apply the min and max? Also, I think adding `maxDateRange` to `start` will allow dates in the future? I tried this locally and was able to select ranges more than 7 days

Reply:
So here's the idea, we want to ensure that the user cannot select more then maxDateRange. The only possible way to select a date range is by 
1. clicking on date to set the start/end date
2. Clicking on another date to set the start/end date.

After step 1. the component should prevent the second date from being more then maxDateRange away. To detect if we are after step 1, we check if start and end are the same.

Reply:
There might be a better way to detect 1. But I'm not aware of that

Reply:
Ah, so, this is basically checking for whether the end date for the range has been chosen? In that case, maybe Vu will know, it'd be nice to make the check more explicit. If not that, maybe a comment to clarify what's happening?